Composer/packages/client/src/components/Header.tsx

```diff
@@ -167,9 +167,16 @@ export const Header = () => {
   } = useLocation();
 
   useEffect(() => {
-    // hide it on the /home page, but make sure not to hide on /bot/stuff/home in case someone names a dialog "home"
-    const hideCondition = !pathname.endsWith('/home') || pathname.includes('/bot/');
-    setStartBotsWidgetVisible(hideCondition);
+    /**
+     * hide it on the /home /projects/* page, show it in /bot/*
+     * home: /home
+     * create: /projects/create
+     * open: /projects/open
+     * save as: /projects/{projectId}/{botName}/save
+     */
+    const hideCondition =
```

Comment:
@srinaath @zhixzhan This solution of checking the current path seems to be flaky, can we decide to show/hide the control based on our app state instead? for example, if root project id has a value or something of that sort?

Reply:
I agree. When i first wrote it i had one pathname to check. But, currently there is no appstate that signifies this use case. RootBotId is set even in homepage

Reply:
I made some updates, created a new `activePageMode` state in global.

Reply:
either directly judging on url patterns, or creating a hook to encapsulate that logic